src/content/1.7/modules/concepts/controllers/admin-controllers/tabs.md

```diff
@@ -253,4 +266,94 @@ class example_module_mailtheme extends Module
+
+Here is an example with a Symfony controller (example comes from the `ps_linklist` module), nothing specific in this controller but you will note the security annotation `@AdminSecurity` that uses `request.get('_legacy_controller')` which will make the link between this controller and the routing configuration.
+
+```php
+// yourmodule/src/Controller/Admin/Improve/Design
+
+namespace PrestaShop\Module\LinkList\Controller\Admin\Improve\Design;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\ModuleActivated;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class LinkBlockController.
+ *
+ * @ModuleActivated(moduleName="ps_linklist", redirectRoute="admin_module_manage")
+ */
+class LinkBlockController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function listAction(Request $request)
+    {
+        //Get hook list, then loop through hooks setting it in in the filter
+        ...
+
+        return $this->render('@Modules/ps_linklist/views/templates/admin/link_block/list.html.twig', [
+            'grids' => $presentedGrids,
+            'enableSidebar' => true,
+            'layoutHeaderToolbarBtn' => $this->getToolbarButtons(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+}
+```
+
+Now here is the routing configuration, we can see the `_legacy_controller` option, that will be used by the controller, is present with a value of `AdminLinkWidget` which will be used as our **class_name** for the tab.
+
+```yaml
+# yourmodule/config/routes.yml
+admin_link_block_list:
+  path: /link-widget/list
+  methods: [GET]
+  defaults:
+    _controller: 'PrestaShop\Module\LinkList\Controller\Admin\Improve\Design\LinkBlockController::listAction'
+    # _legacy_controller is used to manage permissions
+    _legacy_controller: AdminLinkWidget
+    # No need for _legacy_link in this case
+```
+
+Finally here is the `$tabs` property used for automatic registration, it still needs a `class_name` field that will be used for permission checking, it will also be used to create the default `AUTHORIZATION_ROLES` related to this **class_name**.
+
+```php
+// yourmodule/ps_linklist.php
+use Language;
+
+class Ps_Linklist extends Module
+{
+    public function __construct() {
+        ...
+        $tabNames = [];
+        foreach (Language::getLanguages(true) as $lang) {
+            $tabNames[$lang['locale']] = $this->trans('Link List', array(), 'Modules.Linklist.Admin', $lang['locale']);
```

Comment:
Again, id_lang
```suggestion
            $tabNames[$lang['id_lang']] = $this->trans('Link List', array(), 'Modules.Linklist.Admin', $lang['locale']);
```
Also, note that this won't work unless all the languages have been loaded in the translator beforehand...

Reply:
Yep, indeed but that's the best we can propose Unless your fix for tab translation is available in 177?

Although I tried this code with two languages (en and fr) and it worked fine (the wordings were in the core catalog maybe that helped)

Reply:
https://github.com/PrestaShop/ps_linklist/pull/87 Tested in this PR

Reply:
You do not have to loop on each language, the Tab register will handle it for you.